vault/resource_kubernetes_secret_backend.go

```diff
@@ -0,0 +1,178 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/util"
+)
+
+const (
+	fieldKubernetesHost    = "kubernetes_host"
+	fieldKubernetesCACert  = "kubernetes_ca_cert"
+	fieldServiceAccountJWT = "service_account_jwt"
+	fieldDisableLocalCAJWT = "disable_local_ca_jwt"
+)
+
+func kubernetesSecretBackendResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: kubernetesSecretBackendCreateUpdate,
+		ReadContext:   kubernetesSecretBackendRead,
+		UpdateContext: kubernetesSecretBackendCreateUpdate,
+		DeleteContext: kubernetesSecretBackendDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: getKubernetesSecretsMountSchema(),
+	}
+}
+
+func getKubernetesSecretsMountSchema() map[string]*schema.Schema {
+	s := getMountSchema("type")
```

Comment:
This ended up being an easy way to get all of the mount config and tuning options without rewriting code. Happy to consider a different approach if you all would prefer that. It would be nice to have some common mount config code to share for secrets engines.

Reply:
I like that we are including the mount creation with this new resource, normally when provisioning a secret's engine the operator would need to provide `vault_mount` resource.

Reply:
I actually noticed the opposite on the secrets engines that I looked at as a reference. For example, the following secret engine resources include mount creation/tuning directly (i.e., do not require a `vault_mount` resource): aws, azure, consul, gcp, nomad, tfc. 

I do see a couple of them that require a `vault_mount` resource though.

Reply:
ah, maybe I was thinking of the database secrets engines.